numpy/ma/tests/test_extras.py

```diff
@@ -1435,6 +1435,24 @@ def test_setxor1d(self):
         #
         assert_array_equal([], setxor1d([], []))
 
+    def test_isin(self):
+        # the tests for in1d cover most of isin's behavior
+        # if in1d is deprecated, would need to change those tests to test
+        # isin instead.
+        a = np.arange(24).reshape([2, 3, 4])
+        mask = np.zeros([2, 3, 4])
+        mask[1, 2, 0] = 1
+        a = array(a, mask=mask)
+        #masked:   N   Y   N   Y  N  Y   N   Y   N
+        b = array([0, 10, 20, 30, 1, 3, 11, 22, 33],
+                  mask=[0, 1, 0, 1, 0, 1, 0, 1, 0])
+        ec = zeros((2, 3, 4), dtype=bool)
+        ec[0, 0, 0] = True
+        ec[0, 0, 1] = True
+        ec[0, 2, 3] = True
+        c = isin(a, b)
+        assert_array_equal(c, ec)
```

Comment:
Can we have a test here for `type(c)`? Should it be `MaskedArray` or `array`.

Also, a test of what happens if you call `np.isin` instead of `np.ma.isin` might be interesting as well.